libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -70,6 +69,39 @@ public void defaultEngines_didGetInitialized() throws Exception {
     assertNotNull(navigation.getOffRouteEngine());
   }
 
+  @Test
+  public void offRouteEngine_doesNotGetInitializedWithFalseOption() throws Exception {
+    MapboxNavigationOptions options = MapboxNavigationOptions.builder()
+      .enableOffRouteDetection(false)
+      .build();
+    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options, mock(NavigationTelemetry.class),
+      mock(LocationEngine.class));
+    assertNull(navigation.getOffRouteEngine());
+    initDefaultNavigation();
```

Comment:
Why is `initDefaultNavigation()` needed?

👀 [FIRST](https://hackernoon.com/test-f-i-r-s-t-65e42f3adc17)